src/py/flwr/common/message.py

```diff
@@ -391,44 +507,95 @@ def __repr__(self) -> str:
         )
         return f"{self.__class__.__qualname__}({view})"
 
-    def _limit_message_res_ttl(self, message: Message) -> None:
-        """Limit the TTL of the provided Message to not exceed the expiration time of
-        this Message it replies to.
 
-        Parameters
-        ----------
-        message : Message
-            The reply Message to limit the TTL for.
-        """
-        # Calculate the maximum allowed TTL
-        max_allowed_ttl = (
-            self.metadata.created_at + self.metadata.ttl - message.metadata.created_at
+def make_message(
+    metadata: Metadata, content: RecordSet | None = None, error: Error | None = None
```

Comment:
```suggestion
    metadata: Metadata, content: Union[RecordSet, None] = None, error: Union[Error, None] = None
```

I believe using the pipe `|` for union types is introduced in Python 3.10. Until we drop support for Python 3.9, we still have to use `Union`.

Reply:
Resolving based on `from __future__ import annotations`